django_cloud_tasks/models.py

```diff
@@ -0,0 +1,133 @@
+# pylint: disable=no-member
+from datetime import datetime
+from typing import Optional, Dict, List
+from django.db import transaction, models
+from django.db.models import QuerySet, Q
+from django.apps import apps
+from django_cloud_tasks import tasks, serializers
+
+
+class Pipeline(models.Model):
+    name = models.CharField(max_length=100)
+
+    def start(self):
+        routines = self.routines.filter(Q(dependent_routines__id__isnull=True) & ~Q(status=Routine.Statuses.COMPLETED))
+        for routine in routines:
+            routine.enqueue()
+
+    def revert(self):
+        routines = self.routines.filter(Q(next_routines__id__isnull=True) & ~Q(status=Routine.Statuses.REVERTED))
```

Comment:
This query is trickier than this: routines that are REVERTING, PEDING or RUNNING should not be triggered.

Reply:
This is not a big deal, cause the `revert` method will handle based on its status.

Reply:
I don't think so, there's a huge concurrency with these statuses.
We cant try, but I'd be great to leave `TODO` all over to places you know there's still work to be done.